crates/bevy_math/src/bounding/bounded2d.rs

```diff
@@ -0,0 +1,284 @@
+        }
+    }
+
+    #[inline(always)]
+    fn padded(&self, amount: Self::Padding) -> Self {
+        let b = Self {
+            min: self.min - amount.0,
+            max: self.max + amount.1,
+        };
+        debug_assert!(b.min.x <= b.max.x && b.min.y <= b.max.y);
+        b
+    }
+
+    #[inline(always)]
+    fn shrunk(&self, amount: Self::Padding) -> Self {
+        let b = Self {
+            min: self.min + amount.0,
+            max: self.max - amount.1,
+        };
+        debug_assert!(b.min.x <= b.max.x && b.min.y <= b.max.y);
+        b
+    }
+}
+
+#[test]
+fn test_aabb2d_center() {
+    let aabb = Aabb2d {
+        min: Vec2::new(-0.5, -1.),
+        max: Vec2::new(1., 1.),
+    };
+    assert!((aabb.center() - Vec2::new(0.25, 0.)).length() < 0.0001);
+    let aabb = Aabb2d {
+        min: Vec2::new(5., -10.),
+        max: Vec2::new(10., -5.),
+    };
+    assert!((aabb.center() - Vec2::new(7.5, -7.5)).length() < 0.0001);
+}
+
+#[test]
+fn test_aabb2d_area() {
+    let aabb = Aabb2d {
+        min: Vec2::new(-1., -1.),
+        max: Vec2::new(1., 1.),
+    };
+    assert!((aabb.visible_area() - 4.).abs() < 0.0001);
+    let aabb = Aabb2d {
+        min: Vec2::new(0., 0.),
+        max: Vec2::new(1., 0.5),
+    };
+    assert!((aabb.visible_area() - 0.5).abs() < 0.0001);
+}
+
+#[test]
+fn test_aabb2d_contains() {
+    let a = Aabb2d {
+        min: Vec2::new(-1., -1.),
+        max: Vec2::new(1., 1.),
+    };
+    let b = Aabb2d {
+        min: Vec2::new(-2., -1.),
+        max: Vec2::new(1., 1.),
+    };
+    assert!(!a.contains(&b));
+    let b = Aabb2d {
+        min: Vec2::new(-0.25, -0.8),
+        max: Vec2::new(1., 1.),
+    };
+    assert!(a.contains(&b));
+}
+
+#[test]
+fn test_aabb2d_merged() {
+    let a = Aabb2d {
+        min: Vec2::new(-1., -1.),
+        max: Vec2::new(1., 0.5),
+    };
+    let b = Aabb2d {
+        min: Vec2::new(-2., -0.5),
+        max: Vec2::new(0.75, 1.),
+    };
+    let merged = a.merged(&b);
+    assert!((merged.min - Vec2::new(-2., -1.)).length() < 0.0001);
+    assert!((merged.max - Vec2::new(1., 1.)).length() < 0.0001);
+}
+
+#[test]
+fn test_aabb2d_padded() {
+    let a = Aabb2d {
+        min: Vec2::new(-1., -1.),
+        max: Vec2::new(1., 1.),
+    };
+    let padded = a.padded((Vec2::ONE, Vec2::Y));
+    assert!((padded.min - Vec2::new(-2., -2.)).length() < 0.0001);
+    assert!((padded.max - Vec2::new(1., 2.)).length() < 0.0001);
+}
+
+#[test]
+fn test_aabb2d_shrunk() {
+    let a = Aabb2d {
+        min: Vec2::new(-1., -1.),
+        max: Vec2::new(1., 1.),
+    };
+    let shrunk = a.shrunk((Vec2::ONE, Vec2::Y));
+    assert!((shrunk.min - Vec2::new(-0., -0.)).length() < 0.0001);
+    assert!((shrunk.max - Vec2::new(1., 0.)).length() < 0.0001);
+}
+
+/// A bounding circle
+pub struct BoundingCircle {
+    /// The center of the bounding circle
+    pub center: Vec2,
+    /// The radius of the bounding circle
+    pub radius: f32,
```

Comment:
Maybe this could use the `Circle` primitive? It'd give useful helpers for area, perimeter, diameter etc. that could be used both by users and by methods like `visible_area` (if #10632 gets merged). Same for `BoundingSphere`.

Although if we do this, then there would also need to be a `new` constructor so that you can give just the radius.

Either way is fine for me, but something to consider

Reply:
I think I prefer that design.